sdk/scheduler/src/main/java/com/mesosphere/sdk/scheduler/plan/DefaultPlanScheduler.java

```diff
@@ -129,22 +129,57 @@ public DefaultPlanScheduler(
         return acceptedOffers;
     }
 
-    private void killTasks(PodInstance podInstance) {
+    private void killTasks(PodInstanceRequirement podInstanceRequirement) {
         Collection<TaskInfo> tasks = stateStore.fetchTasks();
```

Comment:
Could do a Map<String, TaskInfo> here to make it clear that task names are expected to be unique (and also simplify the following loop a bit). For a minute I was thinking it'd be possible to have two TaskInfos with the same name which would make this loop a bit crazy (tasks getting killed twice etc).